notebooks/jms_to_smallrye_reactive/06_few_shot_with_analysis_jms-to-reactive-ALL.ipynb

```diff
@@ -0,0 +1,754 @@
+      "\u001b[0;31mValueError\u001b[0m: Unable to parse result"
+     ]
+    }
+   ],
+   "source": [
+    "# model_name = \"gpt-4-1106-preview\"\n",
+    "model_name = \"gpt-3.5-turbo-16k\"\n",
+    "llm = ChatOpenAI(temperature=0.1, model_name=model_name)\n",
+    "\n",
+    "common_rule_names = [\"jms-to-reactive-quarkus-00000\"]\n",
+    "for rule_name in common_rule_names:\n",
+    "    print(f\"Processing rule: {rule_name}\")\n",
+    "    template_args = create_template_args_from_incident(\n",
+    "        cmt_report,\n",
+    "        helloworld_report,\n",
+    "        cmt_diff,\n",
+    "        helloworld_diff,\n",
+    "        ruleset_name,\n",
+    "        rule_name,\n",
+    "    )\n",
+    "    formatted_prompt = template_with_solved_example_and_analysis.format(**template_args)\n",
+    "    prompt = PromptTemplate.from_template(template_with_solved_example_and_analysis)\n",
+    "    chain = LLMChain(llm=llm, prompt=prompt)\n",
+    "    result = chain.run(template_args)\n",
+    "\n",
+    "    src_file_name = template_args[\"src_file_name\"]\n",
+    "    src_file_contents = template_args[\"src_file_contents\"]\n",
+    "    # For comparison, we will look up what the source file looks like from Quarkus branch\n",
+    "    # this serves as a way of comparing to what the 'answer' is that was done manually by a knowledgeable developer\n",
+    "    src_file_from_quarkus_branch = helloworld_diff.get_file_contents(\n",
+    "        src_file_name, \"quarkus\"\n",
+    "    )\n",
+    "\n",
+    "    output_dir = (\n",
+    "        f\"output/{model_name}/helloworldmdb/{ruleset_name}/{rule_name}/few_shot/\"\n",
+    "    )\n",
+    "    write_output_to_disk(\n",
+    "        output_dir,\n",
+    "        formatted_prompt,\n",
+    "        src_file_contents,\n",
+    "        src_file_from_quarkus_branch,\n",
+    "        result,\n",
+    "    )\n",
+    "    print(f\"Completed processing of rule: {rule_name}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Should we consider iterating over fixes one at a time?\n"
```

Comment:
I was contemplating about this a while ago. May be worth discussing this with the team.